src/main/java/com/microsoft/graph/authentication/TokenCredentialAuthProvider.java

```diff
@@ -0,0 +1,67 @@
+package com.microsoft.graph.authentication;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+
+import java.net.URL;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import javax.annotation.Nonnull;
+
+/**
+ * An implementation of the Authentication Provider with Azure-identity
+ */
+public class TokenCredentialAuthProvider extends BaseAuthenticationProvider {
+    /** TokenCredential expected from user */
+    private final TokenCredential tokenCredential;
+    /** Context options which can be optionally set by the user */
+    private final TokenRequestContext context;
+    /** Default scope to use when no scopes are provided */
+    private static final String DEFAULT_GRAPH_SCOPE = "https://graph.microsoft.com/.default";
+    /**
+     * Creates an Authentication provider using a passed in TokenCredential
+     *
+     * @param tokenCredential Credential object inheriting the TokenCredential interface used to instantiate the Auth Provider
+     */
+    public TokenCredentialAuthProvider(@Nonnull final TokenCredential tokenCredential) {
+        this(Arrays.asList(new String[] {DEFAULT_GRAPH_SCOPE}), tokenCredential);
```

Comment:
Why not `Arrays.asList(DEFAULT_GRAPH_SCOPE)`?

Reply:
```suggestion
        this(Arrays.asList(DEFAULT_GRAPH_SCOPE), tokenCredential);
```